### DISCUSSION TITLE
Searching a folder with spaces via bash script

### DISCUSSION THREAD

**Discussion by nehasjains** [MusicPlayerDaemon/MPD#1496] [Q&A]
Installed mpd and mpc on my raspberry pi.

I want to search a folder(with spaces) in my library via mpc using bash script

#!/bin/sh
mpc --host=mypwd@127.0.0.1 --port=9999 ls $1

I am launching the script from terminal as follows
$ ./play.sh Bryan\ Adams

I am getting an error mpd error : No such directory.

Whereas if I run the mpc command directly from terminal as below mpc return the folder contents.

$ mpc --host=mypwd@127.0.0.1 --port=999 ls Bryan\ Adams

The command run from terminal fetches the result but if I run a bash script it gives error.

Kindly guide me how to fix this.

Thanks in advance.

**Comment by jcorporation** [comment]
The argument in your script should be quoted:
```
#!/bin/sh
mpc --host=mypwd@127.0.0.1 --port=9999 ls "$1"
```

**Reply by nehasjains** [reply]
Thanks a lot now its working...

**Reply by nehasjains** [reply]
mpc --host=mypwd@127.0.0.1 --port=9999 ls "$1" > /dev/null

I want to hide the output of the command of the directory is not found so using /dev/null but it's still displaying the output. Any way to hide the output? Also tried -q parameter

**Comment by jcorporation** [comment]
Possible you should also redirect STDERR.

```
#!/bin/sh
mpc --host=mypwd@127.0.0.1 --port=9999 ls "$1" 2>&1 > /dev/null
```

**Reply by nehasjains** [reply]
> Possible you should also redirect STDERR.
> 
> ```
> #!/bin/sh
> mpc --host=mypwd@127.0.0.1 --port=9999 ls "$1" 2>&1 > /dev/null
> ```

Wow this worked.

I am a beginner in linux... any book you recommend 

Really thank you for your kind help

**Reply by jcorporation** [reply]
Searching the internet should help ;)

**Reply by nehasjains** [reply]
Yes I agree but eg in this case I had no idea that there is something like "redirect STDERR". So if you can recommend any good book that would be great.

**Reply by skidoo23** [reply]
Old tutorial for bash [https://tldp.org/LDP/Bash-Beginners-Guide/html/Bash-Beginners-Guide.html] Redirection of errors [https://tldp.org/LDP/Bash-Beginners-Guide/html/Bash-Beginners-Guide.html#sect_08_02_03_02] BTW: bash is not a "linux-only" thing it is Unix ;)

**Reply by nehasjains** [reply]
Thanks a lot for your help
